### DISCUSSION TITLE
Us zxing-cpp by an action

### DISCUSSION THREAD

**Discussion by oehhar** [zxing-cpp/zxing-cpp#478] [Q&A] [answered]
Dear zxing-cpp team,
thank you for this great software!
Sorry for the following dump question with little knowledge on my side.

I want to migrate from zxing (barcode decoder) to zxing-cpp on Android.

Up to now, I use an Android activity "**com.google.zxing.client.android.SCAN**" (using [Androwish](www.androwish.org) ) to scan.
Sample app here [Overview](https://www.e-d-c.info/projekte/decoderinterface.html), [Source code](https://www.e-d-c.info/images/edc/decoderinterface/decoder_interface_demo.tcl).
The zxing App shows a scanning user interface and returns the scanned data.

- Is there a similar way with zxing-cpp?
- Or should I ask the magic Androwish Wizard to include zxing-cpp into Androwish, so I can use it there. There is already similar libraries there (zbar, libdmtx), so this might be a good extension. But then, the image capture user interface is not available and should be added.

Thank you for any answer!

Take care,
Harald

**Comment by axxel** [comment, marked as answer]
I'm not an Android developer myself and gobbled together the current sample application code from existing examples and snippets. I don't know how to compose a proper "activity", let alone make sure that it is compatible/usable from your tcl/tk code. I guess your best bet is to get in touch with Androwish as you suggested. That way, other people might also 'benefit' from this code.

**Reply by oehhar** [reply]
Thank you, Axel, I appreciate!
And happy new year, by the way,
Harald
